bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/handler/InsteonDeviceHandler.java

```diff
@@ -136,18 +136,27 @@ public void initialize() {
         config = getConfigAs(InsteonDeviceConfiguration.class);
 
         scheduler.execute(() -> {
-            if (getBridge() == null) {
+            Bridge bridge = getBridge();
+            if (bridge == null) {
                 String msg = "An Insteon network bridge has not been selected for this device.";
-                logger.warn("{} {}", thing.getUID().getAsString(), msg);
+                logger.debug("{} {}", thing.getUID().getAsString(), msg);
 
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
                 return;
             }
 
+            if (!ThingStatus.ONLINE.equals(bridge.getStatus())) {
+                String msg = String.format("Insteon network is not online");
+                logger.debug("{} {}", thing.getUID().getAsString(), msg);
+
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, msg);
+                return;
```

Comment:
It seems this will prevent it from ever coming **ONLINE**? Also, most likely the bridge is not **OFFLINE** in this scenario, but actually **UNKNOWN**. Otherwise child handlers' `initialize()` method shouldn't be called. As far as I can see the only place setting status to **ONLINE** is at the end of this method. This seems like a timing issue. I haven't seen your logs, but I can imagine something like this happening (before your changes) in order:
- Bridge status: INITIALIZING
- Bridge status: UNKNOWN
- Thing status: INITIALIZING
- Thing status: UNKNOWN
- Thing status: ONLINE
- Bridge status: ONLINE

Since the thing always communicates through the bridge, you need to synchronize online state closely with the bridge, as far as I understand.

Perhaps you could implement the Observer pattern where the child things would register as listeners with the bridge. Then they could get notified when the bridge would come online. This would be needed in the scenario where they are initialized faster than the bridge itself will transition from **UNKNOWN** to **ONLINE**.

Reply:
I've tested this and it works.

Reply:
Can you post logs documenting the status transitions (for bridge and thing) with timestamps - if possible, also logs from before the changes?

Reply:
I don't have logs before the change, but here's a snapshot of the current log:

```
2022-05-23 09:00:43.695 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:network:home' changed from UNINITIALIZED to UNINITIALIZED (HANDLER_MISSING_ERROR)
2022-05-23 09:00:45.648 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:network:home' changed from UNINITIALIZED (HANDLER_MISSING_ERROR) to INITIALIZING
2022-05-23 09:00:46.523 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:network:home' changed from INITIALIZING to UNKNOWN
2022-05-23 09:00:46.654 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:device:home:179387' changed from INITIALIZING to OFFLINE (BRIDGE_OFFLINE): Insteon network is not online
...
2022-05-23 09:00:47.919 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:device:home:5469D5' changed from INITIALIZING to OFFLINE (BRIDGE_OFFLINE): Insteon network is not online
2022-05-23 09:00:51.181 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:network:home' changed from UNKNOWN to ONLINE
2022-05-23 09:00:51.518 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:device:home:179387' changed from OFFLINE (BRIDGE_OFFLINE): Insteon network is not online to ONLINE
...
2022-05-23 09:00:51.909 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:device:home:5469D5' changed from OFFLINE (BRIDGE_OFFLINE): Insteon network is not online to ONLINE
```

What would happen is the devices would go straight from `INITIALIZING` to `ONLINE`

Reply:
Thanks for the clarification. So child thing will track bridge transition to **ONLINE**.